### ISSUE TITLE
Constructs that are unsupported by JModelica

### ISSUE THREAD

**Issue by Mathadon** [Mathadon/modelica_MPC#2]
An overview (updated) of some constructs that are not supported by JModelica/CasADi:
 - if/then else constructs, except when embedded in equations, see #8 
 - certain for loops
 - when/while statements
 - therefore flow junctions are also not supported as far as I know
 - discrete variables
 - external C functions (e.g. tables)
 - algorithm sections?
 
@dhblum do you have the same experience?

**Comment by dhblum** [comment]
Sorry I'm just responding now.

I haven't tested external C functions and have to double check algorithms.  But the others are my experience as well.  In addition not supported:
- for/when/while loops
- tables

Some of these constructs (like loops and if/else) are supported in the instantiation section of the model file, but not the equations section.

**Comment by Mathadon** [comment]
Tables are implemented through C-code, so that makes sense.
Some for loops can be reformulated such that they do work, e.g.:
```
Real[5] d;
for i in 1:5 loop
d[i] = time^i;
end for;
```
use instead:
```
d = {time^i for i in 1:5};
```
Such a for loop is expanded during translation, but that only works for 'for expressions' that do not depend on time-depending variables of course.

My experience is also that using `inline=True` can help to get some functions to work.

**Comment by Mathadon** [comment]
@saroele This can be of interest!

**Comment by saroele** [comment]
Yep, this is indeed a good overview, thanks!  This page can also help in case you run into problems:
http://www.jmodelica.org/api-docs/usersguide/2.0/ch11.html
